js/app.js

```diff
@@ -1,262 +1,269 @@
-    hljs.highlightBlock($('#formatted-resp'));
-  };
-  
-  this.rawResponseBody = function() {
-    self.useFormattedResponseBody(false);
-    self.useRawResponseBody(true);
-  };
-
-  let convertToHeaderObj = function(headersList) {
-    let headerObj = {};
-    $.each(headersList, function(index, header) {
-      headerObj[header.name] = header.value;
-    });
-    return headerObj;
-  }
-}
-
-function parseHeaders(headers) {
-  return headers.trim().split('\n').map((header) => {
-    const i = header.indexOf(':');
-    return {
-      name: header.substr(0, i).trim(),
-      value: header.substr(i + 1).trim(),
+    const parseRequest = (req) => {
+      Resting.requestMethod(req.method);
+      Resting.requestUrl(req.url);
+      Resting.bodyType(req.bodyType);
+      updateBody(req.bodyType, req.body);
     };
-  });
-}
-
-
-function EntryListViewModel(params) {
-  const self = this;
-  self.entryList = params.entryList;
-  
-  self.entryName = ko.observable();
-  self.entryValue = ko.observable();
-  
-  
-  self.add = function() {
-    if(checkValidEntry(self.entryName(),self.entryValue())) {
-      self.entryList.push({name: self.entryName() , value: self.entryValue()});
-      self.entryName('');
-      self.entryValue('');
-    }
+
+    const dataToSend = () => {
+      if (Resting.bodyType() === 'form-data') {
+        return convertToFormData(Resting.formDataParams());
+      }
+
+      if (Resting.bodyType() === 'x-www-form-urlencoded') {
+        return convertToUrlEncoded(Resting.formEncodedParams());
+      }
+
+      return Resting.rawBody().trim();
+    };
+
+    const loadBookmark = (bookmarkIdx) => {
+      const selectedBookmark = Resting.bookmarks()[bookmarkIdx()];
+      if (!selectedBookmark) return false;
+
+      return Resting.parseRequest(selectedBookmark.request);
+    };
+
+    const body = (bodyType) => {
+      if (bodyType === 'form-data') {
+        return Resting.formDataParams();
+      }
+
+      if (bodyType === 'x-www-form-urlencoded') {
+        return Resting.formEncodedParams();
+      }
+
+      return Resting.rawBody();
+    };
+
+    const saveBookmark = () => {
+      const request = makeRequest(
+        Resting.requestMethod(), Resting.requestUrl(),
+        Resting.requestHeaders(), Resting.bodyType(),
+        body(Resting.bodyType()));
+      const bookmark = makeBookmark(new Date().toString(), request);
+      localforage.setItem(bookmark.id, JSON.stringify(bookmark));
+      Resting.bookmarks.push(bookmark);
+    };
+
+    const deleteBookmark = bookmark =>
+      localforage.removeItem(bookmark.id)
+        .then(() =>
+          Resting.bookmarks.remove(bookmark));
+
+    const parseHeaders = headers =>
+      headers.trim().split('\n')
+        .map(header =>
+          header.split(':')
+            .map(h => h.trim()))
+        .map(([name, value]) => ({ name, value }));
+
+    const convertToHeaderObj = headersList =>
+      headersList.reduce((acc, [name, value]) => {
```

Comment:
If I put some header in the request this method invocation throws an exception
TypeError: (destructured parameter) is not iterable.

Same problem with methods:
* convertToUrlEncoded
* convertToFormData